aiida/backends/djsite/db/migrations/0031_remove_legacy_workflows.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,too-few-public-methods
+"""Remove legacy workflow."""
+from __future__ import unicode_literals
+
+from __future__ import absolute_import
+import sys
+import click
+
+# Remove when https://github.com/PyCQA/pylint/issues/1931 is fixed
+# pylint: disable=no-name-in-module,import-error
+from django.core import serializers
+from django.db import migrations
+
+from aiida import settings
+from aiida.backends.djsite.db.migrations import upgrade_schema_version
+from aiida.common import json
+from aiida.cmdline.utils import echo
+
+REVISION = '1.0.31'
+DOWN_REVISION = '1.0.30'
+
+
+def export_workflow_data(apps, _):
+    """Export existing legacy workflow data to a JSON file."""
+    from tempfile import NamedTemporaryFile
+
+    DbWorkflow = apps.get_model('db', 'DbWorkflow')
+    DbWorkflowData = apps.get_model('db', 'DbWorkflowData')
+    DbWorkflowStep = apps.get_model('db', 'DbWorkflowStep')
+
+    count_workflow = DbWorkflow.objects.count()
+    count_workflow_data = DbWorkflowData.objects.count()
+    count_workflow_step = DbWorkflowStep.objects.count()
+
+    # Nothing to do if all tables are empty
+    if count_workflow == 0 and count_workflow_data == 0 and count_workflow_step == 0:
+        return
+
+    if not settings.TESTING_MODE:
+        echo.echo('\n')
+        echo.echo_warning('The legacy workflow tables contain data but will have to be dropped to continue.')
```

Comment:
Maybe you can add another line that also says something like "if you really care, make also sure that you save a DB dump of the current version"

Reply:
You can pre-approve it. I will make sure to add it before merging